examples/showcase/where_is_stereo.py

```diff
@@ -0,0 +1,154 @@
+"""
+=============================================
+Reproducing the "Where is STEREO Today?" plot
+=============================================
+
+How to reproduce the "Where is STEREO Today?" plot.
+
+This example uses capabilities in sunpy to reproduce the plot that can be found
+at `<https://stereo-ssc.nascom.nasa.gov/where.shtml>`__.  This example is not
+written as a tutorial, but there are related tutorials for learning about these
+capabilities:
+
+* :ref:`sphx_glr_generated_gallery_units_and_coordinates_planet_locations.py`
+* :ref:`sphx_glr_generated_gallery_units_and_coordinates_ParkerSolarProbe_trajectory.py`
+
+`astroquery <https://astroquery.readthedocs.io/>`__ needs to be installed.
+"""
+import matplotlib as mpl
+import matplotlib.pyplot as plt
+import numpy as np
+from matplotlib.ticker import MultipleLocator
+
+import astropy.units as u
+
+from sunpy.coordinates import HeliocentricEarthEcliptic, get_body_heliographic_stonyhurst, get_horizons_coord
+from sunpy.time import parse_time
+
+##############################################################################
+# Define the time for the plot as the time when this script is run.
+
+obstime = parse_time('now')
+
+##############################################################################
+# Obtain the locations and trajectories of the various planets and spacecraft.
+
+planets = ['Mercury', 'Venus', 'Earth', 'Mars']
+periods = [88, 225, 365, 687]
+planet_coords = {planet: get_body_heliographic_stonyhurst(planet,
+                                                          obstime + np.arange(period) * u.d)
+                 for planet, period in zip(planets, periods)}
+
+stereo_a = get_horizons_coord('STEREO-A', obstime)
+stereo_b = get_horizons_coord('STEREO-B', obstime)
+
+missions = ['Parker Solar Probe', 'Solar Orbiter', 'BepiColombo']
+mission_labels = {'Parker Solar Probe': 'PSP', 'Solar Orbiter': 'SO', 'BepiColombo': 'BEPICOLOMBO'}
+periods = [100, 180, 120]  # may need to be updated as orbits evolve
+mission_coords = {mission: get_horizons_coord(mission, {'start': obstime,
+                                                        'stop': obstime + period * u.d,
+                                                        'step': '1d'})
+                  for mission, period in zip(missions, periods)}
```

Comment:
```suggestion
# Times to sample each orbit at from today, chosen to span at least a full
# orbit of every body plotted
times = np.arange(1000) * u.day
planets = ['Mercury', 'Venus', 'Earth', 'Mars']
planet_coords = {planet: get_body_heliographic_stonyhurst(planet,
                                                          obstime + times)
                 for planet in planets}

stereo_a = get_horizons_coord('STEREO-A', obstime)
stereo_b = get_horizons_coord('STEREO-B', obstime)

missions = ['Parker Solar Probe', 'Solar Orbiter', 'BepiColombo']
mission_labels = {'Parker Solar Probe': 'PSP', 'Solar Orbiter': 'SO', 'BepiColombo': 'BEPICOLOMBO'}
mission_coords = {mission: get_horizons_coord(mission, {'start': obstime,
                                                        'stop': obstime + times,
                                                        'step': '1d'})
                  for mission in missions}
```

I got very confused about periods for a while here, so would suggest the above to simplify and clarify things.

Reply:
Hmm, I don't like the idea of plotting multiple orbits just to make the code simpler.  It won't matter much for planets, but it could look bad for spacecraft.

An alternative approach that'd be more robust than hard-coding orbital periods, but would mean even more code, is to grab 1000 days as you have done here, but then figure out how many elements make up the first orbit (by looking for when the ecliptic longitude passes the starting point) and then plotting just that first orbit.

Reply:
Ah fair enough, I'm guessing it would look bad because of the dashed lines? In that case hardcoding is probably best, but maybe with a clarifying comment along the lines of `# Orbital period in days` above where `periods` is defined?

Reply:
> Ah fair enough, I'm guessing it would look bad because of the dashed lines?

Not just that, but orbital precession means that successive orbits of spacecraft – most notably BepiColombo – do not lie on top of each other.

I've added a function to automatically determine the first full orbit in a trajectory.